File: src/ShoeMoney.Common/Data/Entities/Order.cs
using ShoeMoney.Enums;

namespace ShoeMoney.Data.Entities;

public class Order
{
  public int Id { get; set; }
  public DateTime OrderDate { get; set; }
  public string? Notes { get; set; }
  public OrderType OrderType { get; set; }
  public string? CompanyName { get; set; }
  public string Contact { get; set; } = "";
  public string? Email { get; set; }
  public string? PhoneNumber { get; set; }

  public Address? ShippingAddress { get; set; }

  public virtual ICollection<OrderItem>? Items { get; set; }
}

File: src/ShoeMoney.Store/src/models/Order.ts
import type { Address } from "./Address";
import type { OrderItem } from "./OrderItem";
import type { OrderStatus } from "./OrderStatus";
import type { OrderType } from "./OrderType";
import type { Payment } from "./Payment";

export interface Order {
  id: number;
  orderDate: string;
  notes: string | null;
  orderType: OrderType;
  companyName: string | null;
  contact: string;
  email: string | null;
  phoneNumber: string | null;
  orderStatus: OrderStatus;
  shippingAddress: Address | null;
  payment: Payment | null;
  items: OrderItem[];
}